vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimInjector.kt

```diff
@@ -50,7 +50,26 @@ interface VimInjector {
   // [FINISHED] Can't be fully moved to vim-engine.
   val enabler: VimEnabler
 
-  // TODO We should somehow state that [OptionServiceImpl] can be used from any implementation
+  /**
+   * Gets an API for consuming only global options
+   *
+   * This function is intended to retrieve global options, not global values for options that are local to buffer, local
+   * to window or global-local - for that, use [OptionService.getOptionValue]. Typical option access should use
+   * [VimInjector.options] and pass in a [VimEditor] for context. This will return local or global values as
+   * appropriate.
+   */
+  fun globalOptions() = optionService.getValueAccessor(null)
```

Comment:
I'd prefer to have these methods as extensions on VimInjector. In this way they won't be a part of the interface and there is no risk that some implementation will override it.

Reply:
I did do that originally, but moved it to an interface method to keep Java consumers simple. This felt too clumsy to write, when the aim of adding the accessor was to simplify the API:

```
final int scrolloff = VimInjectorKt.options(injector, editor).getIntValue(OptionConstants.scrolloff);
```

But we could use static import for the function:

```
final int scrolloff = options(injector, editor).getIntValue(OptionConstants.scrolloff);
```

What do you think?

Reply:
Static imports seem fine to me. Also, taking the fact that we slowly move the code to kt, this inconvenience should be eliminated.

Reply:
We have actually like 10% of java in the sources. We can make a move and convert everything to kt

Reply:
I've just pushed a new commit that replaces the default interface methods with extension methods.